src/tests/test_tls_hybrid_kem_key.cpp

```diff
@@ -0,0 +1,258 @@
+/*
+* (C) 2023 Jack Lloyd
+* (C) 2023 Fabian Albert, René Meusel - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_TLS_13_PQC) && defined(BOTAN_HAS_KYBER) && defined(BOTAN_HAS_DIFFIE_HELLMAN) && \
+   defined(BOTAN_HAS_ECDSA)
+
+   #include <botan/pk_algs.h>
+   #include <botan/internal/hybrid_public_key.h>
+   #include <botan/internal/kex_to_kem_adapter.h>
+   #include <botan/internal/stl_util.h>
+
+namespace Botan_Tests {
+
+namespace {
+
+std::unique_ptr<Botan::Private_Key> kem() {
+   static auto kem_key = Botan::create_private_key("Kyber", Test::rng(), "Kyber-512-r3");
```

Comment:
Why are these test keys `static`? I don't understand.

Reply:
The actual key isn't important for those tests. They just need "some" Kyber key. So this will generate one key and then re-use it to save some time. Now, generating a Kyber key is reasonably fast, so  we could avoid doing this. Or add a comment explaining the idea.

Reply:
Ah I see. I'd somewhat prefer non-static, and a new key is created for each invocation, unless there are performance difficulties as a result. But at the least there should be a comment, since otherwise the why-is-this-static is not so clear.

Reply:
I tried removing the `static` and realized that the tests assume that these functions return the same key with every invocation. I aborted an attempt to refactor the test case to remove this assumption. It appeared that this would result in less clarity and/or added template complexity.

Reply:
:+1: